addon/helpers/cannot.js

```diff
@@ -1,10 +1,12 @@
 import Ember from 'ember';
 
-const { getOwner } = Ember;
+const { getOwner, typeOf } = Ember;
 
 export default Ember.Helper.extend({
   helper: Ember.computed(function() {
-    return getOwner(this).lookup('helper:can');
+    var helper = getOwner(this).lookup('helper:can');
+
+    return typeOf(helper) === 'instance' ? helper : helper.create();
```

Comment:
I didn't necessarily _find_ it from anywhere, but merely figured we could take use of `Ember.typeOf` to distinguish helper `instance`s from `class`es, of which the latter seems to be what gets returned by `.lookup` since Ember 2.10.